[Doc] Explain how `<PasswordInput>` can be used to update a password (#9354)
marmelab/react-admin: docs/PasswordInput.md

Use it like a [`<TextInput>`](./TextInput.md):

```jsx
import { Create, SimpleForm, TextInput, PasswordInput } from 'react-admin';

> const UserCreate = () => (

Reviewer: ```suggestion
export const UserCreate = () => (
```

<PasswordInput source="password" inputProps={{ autocomplete: 'current-password' }} />
```
{% endraw %}

> 

Author: We probably need to explain how to have 2 password inputs, and how to validate that they are equal.

Author: explained.

<PasswordInput source="password" inputProps={{ autocomplete: 'current-password' }} />
```
{% endraw %}

## Validating Identical Passwords

> If you want to validate that the user has entered the same password in two different password inputs, you must use [global validation](./Validation.md#global-validation) at the form level:

Reviewer 2: Not really, validators have access to the other form values, so you could write an `equalToFieldValidator`:

```tsx
const equalToFieldValidator = (source: string) => (value: any, allValues: FormValues) => ...
```

Author: TIL!

```jsx
import { Create, SimpleForm, TextInput, PasswordInput } from 'react-admin';

const validate = values => {
    const errors = {} as any;
    if (!values.name) {
        errors.name = 'Name is required';
    }
    if (!values.email) {
        errors.email = 'Email is required';
    }
    if (!values.password) {
        errors.password = 'Password is required';
    }
    if (values.password && values.password !== values.confirm_password) {
        errors.confirm_password = 'The two password fields must match';
    }
    return errors;
}

> const UserCreate = () => (

Reviewer: ```suggestion
export const UserCreate = () => (
```

    <Create validate={validate}>
        <SimpleForm>
            <TextInput source="name" isRequired />
            <TextInput source="email" isRequired />
            <PasswordInput source="password" isRequired />
            <PasswordInput source="confirm_password" />
        </SimpleForm>
    </Create>
);
```

You can't use the `validate` prop in the Input components, as global and field-level validation are mutually exclusive. To show that an input is required, use the `isRequired` prop instead of `validate`.

## Usage in Edit Views

> You may want to allow users to *update* a password on an existing record. The usual solution to this is to include a `new_password` input in the Edition form. The API will then check if this field is present in the payload, and update the password accordingly.

Reviewer: ```suggestion
You may want to allow users to *update* a password on an existing record. The usual solution to this is to include a `new_password` input in the Edition form. Your API should then check if this field is present in the payload, and update the password accordingly.
```

```jsx
import { Edit, SimpleForm, TextInput, PasswordInput } from 'react-admin';

> const UserEdit = () => (

Reviewer: ```suggestion
export const UserEdit = () => (
```
